warehouse/query-core/src/main/java/datawave/query/util/sortedmap/rfile/KeyValueByteDocumentTransforms.java

```diff
@@ -0,0 +1,78 @@
+package datawave.query.util.sortedmap.rfile;
+
+import java.io.ByteArrayInputStream;
+import java.util.Map;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.commons.collections.keyvalue.UnmodifiableMapEntry;
+
+import datawave.query.attributes.Attribute;
+import datawave.query.attributes.Document;
+import datawave.query.function.deserializer.DocumentDeserializer;
+import datawave.query.function.deserializer.KryoDocumentDeserializer;
+import datawave.query.function.serializer.DocumentSerializer;
+import datawave.query.function.serializer.KryoDocumentSerializer;
+
+public class KeyValueByteDocumentTransforms {
+
+    public static byte[] keyToByte(Key key) {
+        if (key == null) {
+            return null;
+        }
+        return key.getRow().getBytes();
+    }
+
+    public static Key byteToKey(byte[] bytes) {
+        if (bytes == null) {
+            return null;
+        }
+        return new Key(bytes);
+    }
+
+    public static Value documentToValue(Document doc) {
+        if (doc == null) {
+            return null;
+        }
+        DocumentSerializer serializer = new KryoDocumentSerializer(false, true);
+        byte[] document = serializer.serialize(doc);
+        return new Value(document);
+    }
+
+    public static Document valueToDocument(Value value) {
+        if (value == null) {
+            return null;
+        }
+        DocumentDeserializer deserializer = new KryoDocumentDeserializer();
+        Document document = deserializer.deserialize(new ByteArrayInputStream(value.get()));
+        return document;
+    }
```

Comment:
I'm almost 100% certain this is going to be cost prohibitive from a performance perspective. Document serializers/deserializers should be initialized once and reused. Any class using these static methods should instead have a private instance.

Reply:
Nice observation!  I will do as you suggest.

Reply:
also worth keeping in mind that Kryo, backed by KryoDocumentSerializer is NOT thread safe

Reply:
done